src/hooks/useCopyToClipboard.ts

```diff
@@ -0,0 +1,34 @@
+import { useEffect, useState } from 'react';
+
+export const copyToClipboard = async (value: string) => {
+  try {
+    await navigator.clipboard.writeText(value);
```

Comment:
If this is a Promise, can't we use native Promise.catch()?

Reply:
sure we can but is there any issue using async/await

Reply:
Code smell. And avoiding concurrency, IMHO the whole thing can be a chained promise.